File: leetcode_js/leetcode/editor/cn/[1]两数之和.js

//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
	const map = {};
	const array = [];
	for(let i=0;i<nums.length;i++) {
		map[nums[i]] = i;
	}
	for(let i=0;i<nums.length;i++) {
		if(map[target-nums[i]]&&map[target-nums[i]]!==i) {
			array.push(i);
			array.push(map[target-nums[i]]);
			break;
		}
	}
	return array;
};
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode_go/leetcode/editor/cn/[1]两数之和.go

//leetcode submit region begin(Prohibit modification and deletion)
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for index, num := range nums{
		m[num] = index+1
	}
	array := make([]int, 2)
	for i,num := range nums{
		index := m[target-num]
		if m[target-num]!=0&&i!=index-1{
			array[0] = i
			array[1] = index-1
			break
		}
	}
	return array
}
//leetcode submit region end(Prohibit modification and deletion)
